### PR TITLE
[Elastic Agent] Fix broken enrollment command

### LINKED ISSUE #26721: Fleet fails to enroll in 8.0.0-SNAPSHOT version
Using latest 8.0.0-SNAPSHOT version of elastic stack 

`
docker.elastic.co/beats/elastic-agent             8.0.0-SNAPSHOT    913a89b8f631   9 hours ago    571MB
 docker.elastic.co/elasticsearch/elasticsearch     8.0.0-SNAPSHOT    7df40aa3e998   12 hours ago   670MB
 docker.elastic.co/kibana/kibana                   8.0.0-SNAPSHOT    0e35cc0c3784   12 hours ago   1.34GB
`

the fleet server fails to become ready ([fleet-server-logs.zip](https://github.com/elastic/beats/files/6764667/fleet-server-logs.zip)) as it fails to enrol.

In version 7.14.0-SNAPSHOT it was working properly.

Steps to reproduce

1. `elastic-package stack update -v --version=8.0.0-SNAPSHOT`
2. `elastic-package stack up -d -v --version=8.0.0-SNAPSHOT`

[stack up logs.zip](https://github.com/elastic/beats/files/6764710/stack.up.logs.zip)

Fleet server status is DEGRADED.

`docker exec -ti 6ce7e7c23389 /bin/bash`

`bash-4.2$ curl http://localhost:8220/api/status
{"name":"fleet-server","status":"DEGRADED"}`

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

The proxy work broke the `enroll` command that prevents enrollment from working when running Fleet Server with default options.

## Why is it important?

Otherwise the enrollment command will not work also breaking the container.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #26721

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go [hunk 1]
```diff
@@ -419,7 +420,7 @@ func (c *enrollCmd) enrollWithBackoff(ctx context.Context, persistentConfig map[
 			break
 		}
 		backExp.Wait()
-		c.log.Info("Retrying to enroll...")
+		c.log.Infof("Retrying to enrollment to URL: %s", c.client.URI())
```
Comment (mtojek) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:423]: nit: to enroll/Retrying enrollment

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go [hunk 2]
```diff
@@ -200,6 +200,15 @@ func (c *enrollCmd) Execute(ctx context.Context) error {
 	// will communicate to the internal fleet server on localhost only.
 	// Connection setup should disable proxies in that case.
 	localFleetServer := c.options.FleetServer.ConnStr != ""
+	if localFleetServer {
+		token, err := c.fleetServerBootstrap(ctx)
+		if err != nil {
+			return err
```
Comment (mtojek) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:206]: nit: wrap an error with some context?

**Review by mtojek — COMMENTED** [review]
@ruflin as we're talking about running integration tests in terms of Package Registry -
this PR is a good candidate for enabling basic integration tests to make sure if API isn't broken.

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go (hunk 2 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:206]: errors are already wrapped with information inside of the `fleetServerBootstrap`.

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go (hunk 1 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go]: fixed

**Review by michalpristas — APPROVED** [review]
I like updated log verbosity

**Comment by michalpristas** [comment]
tested ok

**Review by urso — APPROVED** [review]

**Comment by ruflin** [comment]
@blakerouse How do we prevent these kind of things to break again? It seems a pretty fundamental thing that was broken and I assume should be uncovered by automated tests even before merging?

**Comment by blakerouse** [comment]
@ruflin This should have been caught by the e2e testing suite. It was not because from what I can tell it seems that the e2e testing suite is not actually running with the package builds from the repository.

You can see from the e2e testing logs in this PR that the log line `Retrying to enroll...` was still shown, that means that the change from this PR was not actually tested by the e2e suite (I assume it is using the last built snapshot), as the log line was change to `Starting enrollment to URL: {url}`.

This needs to be fixed in the e2e repository and the e2e tests should probably now always run. Being that Elastic Agent is going GA things like this should not break.

**Comment by ruflin** [comment]
@blakerouse Can you follow up on this to make sure it is fixed wherever it needs fixing?

**Comment by blakerouse** [comment]
@ruflin https://github.com/elastic/e2e-testing/issues/1323

**Pull request merged**